src/twisted/conch/ssh/keys.py

```diff
@@ -1201,25 +1208,194 @@ def toString(self, type, extra=None):
         @param extra: Any extra data supported by the selected format which
             is not part of the key itself.  For public OpenSSH keys, this is
             a comment.  For private OpenSSH keys, this is a passphrase to
-            encrypt with.
+            encrypt with.  (Deprecated; use C{comment} or C{passphrase} as
+            appropriate instead.)
         @type extra: L{bytes} or L{unicode} or L{None}
 
+        @param subtype: A subtype of the requested C{type} to emit.  Only
+            supported for private OpenSSH keys, for which the currently
+            supported subtypes are C{'PEM'} and C{'v1'}.  If not given, an
+            appropriate default is used.
+        @type subtype: L{str} or L{None}
+
+        @param comment: A comment to include with the key.  Only supported
+            for OpenSSH keys.
+        @type comment: L{bytes} or L{unicode} or L{None}
+
+        @param passphrase: A passphrase to encrypt the key with.  Only
+            supported for private OpenSSH keys.
+        @type passphrase: L{bytes} or L{unicode} or L{None}
+
         @rtype: L{bytes}
         """
-        if isinstance(extra, unicode):
-            extra = extra.encode("utf-8")
+        if extra is not None:
+            # Compatibility with old parameter format.
+            if self.isPublic():
+                comment = extra
+            else:
+                passphrase = extra
+        if isinstance(comment, unicode):
+            comment = comment.encode("utf-8")
+        if isinstance(passphrase, unicode):
+            passphrase = passphrase.encode("utf-8")
```

Comment:
Can you explain what you mean by "normalize" in this context?

Reply:
I mean like NFKC/NFKD normalization, as per https://pages.nist.gov/800-63-3/sp800-63b.html#memsecretver .

It's fine to leave this for a follow-up though, since as I noted it's not new.

Reply:
Thanks.  I've followed up on this in https://github.com/twisted/twisted/pull/1201.